src/uproot/writing/interpret.py

```diff
@@ -0,0 +1,306 @@
+            x_errors_high (float or left unspecified): Asymmetrical upper error values for corresponding x value
+            y_errors_low (float or left unspecified): Asymmetrical lower error values for corresponding y value
+            y_errors_high (float or left unspecified): Asymmetrical upper error values for corresponding y value
+            (other column names will be ignored!)
+        title (str): Title of the histogram.
+        xAxisLabel (str): Label of the X axis.
+        yAxisLabel (str): Label of the Y axis.
+        minY (None or float): Minimum value on the Y axis to be shown, if set to None then minY=min(y)
+        maxY (None or float): Maximum value on the Y axis to be shown, if set to None then maxY=max(y)
+        lineColor (int): Line color. (https://root.cern.ch/doc/master/classTAttLine.html)
+        lineStyle (int): Line style.
+        lineWidth (int): Line width.
+        markerColor (int): Marker color. (https://root.cern.ch/doc/master/classTAttMarker.html)
+        markerStyle (int): Marker style.
+        markerSize (float): Marker size.
+
+    WARNING! This function only works for TGraph, because serialization of TGraphErrors and TGraphAsymmErrors is not implemented yet.
+
+    Function that converts DataFrame into TGraph, TGraphErrors or TGraphAsymmErros based on the specified DataFrame columns.
+    When all error columns are unspecified, detected object is TGraph.
+    When x_errors, y_errors are specified, detected object is TGraphErrors.
+    When x_errors_low, x_errors_high, y_errors_low, y_errors_high are specified, detected object is TGraphAsymmErrors.
+    Note that both {x_errors, x_errors} need to be specified or set to None.
+    The same rule applies {to x_errors_low, x_errors_high, x_errors_low, x_errors_high}.
+    Also can't specify {x_errors, y_errors} and {x_errors_low, x_errors_high, y_errors_low, y_errors_high} at the same time.
+    """
+
+    x = numpy.array(df["x"]) if df.get("x", None) is not None else None
+    y = numpy.array(df["y"]) if df.get("y", None) is not None else None
+    x_errors = (
+        numpy.array(df["x_errors"]) if df.get("x_errors", None) is not None else None
+    )
+    y_errors = (
+        numpy.array(df["y_errors"]) if df.get("y_errors", None) is not None else None
+    )
+    x_errors_low = (
+        numpy.array(df["x_errors_low"])
+        if df.get("x_errors_low", None) is not None
+        else None
+    )
+    x_errors_high = (
+        numpy.array(df["x_errors_high"])
+        if df.get("x_errors_high", None) is not None
+        else None
+    )
+    y_errors_low = (
+        numpy.array(df["y_errors_low"])
+        if df.get("y_errors_low", None) is not None
+        else None
+    )
+    y_errors_high = (
+        numpy.array(df["y_errors_high"])
+        if df.get("y_errors_high", None) is not None
+        else None
+    )
+
+    return _to_TGraph(
```

Comment:
I'm not sure why this function has been split into a public part and a private part, but it's fine.

Reply:
When I started to make this function I wanted all the arrays to be passed as arguments, but later I just changed it to accept dict like object.